docs/stability-wasi-proposals-support.md

```diff
@@ -0,0 +1,38 @@
+# WASI Proposals Support
+
+Wasmtime has initial support for some WASI proposals to assist in their
+development as well as allowing users to more easily provide feedback by testing
+them. This is somewhat of a work-in-progress because there aren't a lot of WASI
+proposals at this time.
+
+Currently supported WASI proposals are:
+
+* [The base WASI specification](https://github.com/webassembly/wasi) - the
+  `snapshot` directory is supported by the [`wasmtime-wasi`
+  crate](https://docs.rs/wasmtime-wasi/0.21.0/wasmtime_wasi/)
+
+* [The wasi-nn proposal](https://github.com/WebAssembly/wasi-nn) - the
+  `ephemeral` directory with the wasi-nn APIs.
+
+Wasmtime strives to support WASI proposals in-tree to assist with
+maintainability and provide easier to support to users for easier feedback. In
+doing so, however, we have a few guidelines around adding support for a WASI
+proposal to Wasmtime:
```

Comment:
Would it make sense to require these to have an RFC filed for them? I think that'd be good practice, and now that we're starting to actually use RFCs, it seems like there's not too much blocking that.

Reply:
👍 I discussed wasi-nn multiple times on Zulip and the Wasmtime meetings; an RFC would have hopefully recorded all of that in one place.